src/Controller/Controller.php

```diff
@@ -386,17 +422,113 @@ public function __set($name, $value)
     }
 
     /**
+     * Returns the controller name.
+     *
+     * @return string
+     * @since 3.6.0
+     */
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    /**
+     * Sets the controller name.
+     *
+     * @param string $name Controller name.
+     * @return $this
+     * @since 3.6.0
+     */
+    public function setName($name)
+    {
+        $this->name = $name;
+
+        return $this;
+    }
+
+    /**
+     * Returns the plugin name.
+     *
+     * @return string|null
+     * @since 3.6.0
+     */
+    public function getPlugin()
```

Comment:
I don't think we should `getPlugin()`, `setPlugin()`. Just make people use `$this->request->getParam('plugin')` instead.

Reply:
I tried this but some tests were failing. It looked to me that there were some cases where different state od request plugin param and controller plugin name was expected.

Reply:
Okay, that's sad :disappointed:.